.github/scripts/main/main.sh

```diff
@@ -52,6 +52,82 @@ make install
 export PATH="$PREFIX/bin:$PATH"
 opam --version
 
+if [ "$OPAM_DOC" = "1" ]; then
+  rm -rf src_ext/
+  make -C doc html man-html pages
+
+  if [ "$GITHUB_EVENT_NAME" = "pull_request" ]; then
+    . .github/scripts/common/hygiene-preamble.sh
+    diff="git diff $BASE_REF_SHA..$PR_REF_SHA"
+    set +e
+    files="`$diff --name-only --diff-filter=A | grep 'src/.*mli'`"
+    set -e
+    if [ -n "$files" ]; then
+      echo '::group::new module added - checking it'
+      if $diff --name-only --exit-code -- doc/index.html ; then
+        echo '::error new module added but index not updates'
+        echo "$files"
+        exit 3
+      fi
+      echo '::endgroup::new module added - checking it'
+    else
+      echo 'No new module added'
+    fi
+  fi
+
+  files="
+    doc/index.html
+    doc/html/index.html
+    doc/man-html/index.html
+
+    doc/html/opam-admin/index.html
+    doc/html/opam-core/index.html
+    doc/html/opam-devel/index.html
+    doc/html/opam-format/index.html
+    doc/html/opam/index.html
+    doc/html/opam-installer/index.html
+    doc/html/opam-repository/index.html
+    doc/html/opam-solver/index.html
+    doc/html/opam-state/index.html
+
+    doc/pages/About.html
```

Comment:
This seems a bit brittle - if a page is added, you have to remember that it's been updated (and similarly it's checking for "generation" of committed files. Wouldn't this be OK: `$(ls doc/pages/*.md | sed -e 's/\.md/.html/')`?

Reply:
good point